substrate/frame/contracts/proc-macro/src/lib.rs

```diff
@@ -206,9 +207,25 @@ impl HostFn {
 					}
 					not_deprecated = false;
 				},
+				"mutable" => {
```

Comment:
Functions shout be `mutating` by default. It is safer to assume that and only flag functions where we know they are read only. So the attribute should be `read_only`. What do you think?

Reply:
I would need to add `#[read_only]` to `call_chain_extension` and `call` to avoid generating the `read-only` check. I believe this would be misleading, as these functions may or may not be `read-only`, depending on the implementation.

Reply:
That is a good point. Didn't think of that. Can you rename to `mutating`?  That sounds better to me as the function itself is not `mutable` like a variable. It `mutates`.

Reply:
Sure, `mutating` is fine

Reply:
done